crates/ruff_linter/resources/test/fixtures/pylint/dict_iter_missing_items.py

```diff
@@ -0,0 +1,23 @@
+d = {1: 1, 2: 2}
+d_tuple = {(1, 2): 3, (4, 5): 6}
+l = [1, 2]
+s1 = {1, 2}
+s2 = {1, 2, 3}
+
+# Errors
+for k, v in d:
+    pass
+
+# False positive, since the keys are all tuples this is valid
+for a, b in d_tuple: 
+    pass
```

Comment:
I wonder if we could avoid raising this false positive. I haven't explored this but maybe we could use `binding.statement(model)` to get the statement where the binding is defined and then we could check if the RHS is a dictionary with a 2-value tuple key.

Reply:
I played around with this and came up with something, as well as two additional test cases to make sure it "works".

It's in the last commit but I'm really not sure about the style (this `rust` thing is kind a new to me). So please tell me how to format it to be as expected (feel free to push to the branch).

Reply:
Thanks! Looks good. I've extracted out the logic to a function to avoid having nested if conditions.

Reply:
Thanks, that looks much better!